dbms/src/DataStreams/FilterTransformAction.cpp

```diff
@@ -99,61 +101,31 @@ bool FilterTransformAction::transform(Block & block, FilterPtr child_filter)
         return true;
     }
 
-    IColumn::Filter * filter;
-    ColumnPtr filter_holder;
-
     if (constant_filter_description.always_true)
     {
-        if (child_filter)
-            filter = child_filter;
-        else
-            return true;
+        if (return_filter)
+            res_filter = nullptr;
+        return true;
     }
     else
     {
         FilterDescription filter_and_holder(*column_of_filter);
-        filter = const_cast<IColumn::Filter *>(filter_and_holder.data);
-        filter_holder = filter_and_holder.data_holder;
-
-        if (child_filter)
-        {
-            /// Merge child_filter
-            std::transform(filter->cbegin(), filter->cend(), child_filter->cbegin(), filter->begin(), [](const UInt8 a, const UInt8 b) { return a > 0 && b != 0; });
-        }
-    }
-
-    /** Let's find out how many rows will be in result.
-      * To do this, we filter out the first non-constant column
-      *  or calculate number of set bytes in the filter.
-      */
-    size_t first_non_constant_column = 0;
-    for (size_t i = 0; i < columns; ++i)
-    {
-        if (!block.safeGetByPosition(i).column->isColumnConst())
-        {
-            first_non_constant_column = i;
-
-            if (first_non_constant_column != filter_column)
-                break;
-        }
```

Comment:
Why remove this?

Reply:
These code is used to caculate `filtered_row`, the idea is:

- find first non constant column, filter, and count how many row left.

So, there is no need to `countBytesInFilter`.

But now we need `filtered_row` before filter, to handle `filtered_row == 0`, so always call `countBytesInFilter`.

Reply:
This may degrades performance, but `countBytesInFilter` is fast, and the main bottleleck of `FilterBlockInputStream` is `expression->execute(block)` and `filter`, so it is acceptable.

Reply:
> These code is used to caculate `filtered_row`, the idea is:
> 
> * find first non constant column, filter, and count how many row left.
> 
> So, there is no need to `countBytesInFilter`.
> 
> But now we need `filtered_row` before filter, to handle `filtered_row == 0`, so always call `countBytesInFilter`.

Get it